2-mandatory/3-shopping-list.js

```diff
@@ -19,7 +19,14 @@ let pantry = {
 };
 
 function createShoppingList(recipe) {
-  // write code here
+
+  const missingArr = recipe.ingredients.filter(eachIngredients =>
+    !pantry.fridgeContents.includes(eachIngredients) && !pantry.cupboardContents.includes(eachIngredients)
+  );
+  return {
+    name: recipe.name,
+    items: missingArr
+  }
 }
```

Comment:
Good job @farzaneh-haghani  , well done.
One more thing you can do here, is maybe  use `concat` function which will merge two arrays and put all of their values in one array and in that array you can just check once if the pantry has the fridge and cup board contents.
`let allContents = pantry.fridgeContents.concat(pantry.cupboardContents)
  let missArr = recipe.ingredients.filter(eachIngredients => !allContents.includes(eachIngredients)) ....`

Reply:
Thank you @ellietms 👍  I will update it.